### PR TITLE
RemoveContainer in Runtime interface

### PR DESCRIPTION
**Pull request by ronnielai**
- Added a DeleteContainer method in Runtime interface
- Implemented DeleteContainer for docker
#25239

### CODE REVIEW

File: pkg/kubelet/container/runtime.go [hunk 1]
```diff
@@ -119,6 +119,8 @@ type Runtime interface {
 	// stream the log. Set 'follow' to false and specify the number of lines (e.g.
 	// "100" or "all") to tail the log.
 	GetContainerLogs(pod *api.Pod, containerID ContainerID, logOptions *api.PodLogOptions, stdout, stderr io.Writer) (err error)
+	// Remove a container. If the container is still running, an error is returned.
```
Comment (vishh) [inline pkg/kubelet/container/runtime.go]: nit: Should we provide a pod-context? specifically in the case of `rkt`? cc @yifan-gu

File: pkg/kubelet/container/runtime.go [hunk 2]
```diff
@@ -119,6 +119,8 @@ type Runtime interface {
...
+	RemoveContainer(containerID ContainerID) error
```
Comment (vishh) [inline pkg/kubelet/container/runtime.go]: Also should this be `RemoveDeadContainer` or `RemoveContainer`?

Comment (vishh) [inline pkg/kubelet/container/runtime.go]: Do we care if the container is running or dead? From an API perspective, making this generic is better. WDYT?

File: pkg/kubelet/container/runtime.go (hunk 1 above)
Comment (vishh) [inline pkg/kubelet/container/runtime.go]: It looks like `rkt` can already handle containerID like in [this](https://github.com/kubernetes/kubernetes/pull/28627/files#diff-a53fa76e941e0bdaee26dcbc435ad2ffR110) case. So it should be OK I think.

File: pkg/kubelet/container/runtime.go (hunk 2 above)
Comment (ronnielai) [inline pkg/kubelet/container/runtime.go]: Renamed it to DeleteContainer to fit the semantics of the new runtime interface.

**Pull request merged**